Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -988,20 +990,34 @@ static void observeSelectors(CSSParserObserverWrapper& wrapper, CSSParserTokenRa
     wrapper.observer().endRuleHeader(wrapper.endOffset(originalRange));
 }
 
-RefPtr<StyleRule> CSSParserImpl::consumeStyleRule(CSSParserTokenRange prelude, CSSParserTokenRange block)
+RefPtr<StyleRule> CSSParserImpl::consumeStyleRule(CSSParserTokenRange prelude, CSSParserTokenRange block, StyleRule* parentRule)
```

Comment:
Maybe we should we be passing a `std::optional<Ref<StyleRule>>` instead?

Reply:
There is a small overhead by using a ref-counted pointer instead of raw pointer. In this kind of nested calls, we know for sure that the pointer will exists at least until we come back to the caller. But maybe it's safer/cleaner to use a `RefPtr` anyway.

Reply:
There is a small overhead by using a ref-counted pointer instead of raw pointer. In this kind of nested calls, we know for sure that the pointer will exists at least until we come back to the caller. But maybe it's safer/cleaner to use a `RefPtr` anyway.